state_machine/applications/flight/lib/mathutils.py

```diff
@@ -52,3 +52,18 @@ def quaternion_to_left_matrix(q):
          [qv,              dr]])
 
     return M
+
+def quaternion_to_rotation_matrix(q):
+    """Converts a scalar-first unit quaternion into the rotation matrix Qv = qvq+"""
```

Comment:
This doesn't necessarily have to be in this file, but at some point (in the future) it would be nice to document the math/physics behind this. To be very honest, I don't fully understand kalman filters because I haven't done the same amount of reading as you on the topic. That being said, it would be nice for a rookie to be able to have some basic idea of what's going on.

Reply:
Honestly, that's not really within the scope of this project. 
I don't think it's the job of the documentation to explain what a Kalman Filter is (this is *very* easily googlable). 

However, I will describe what kind of Kalman Filter this is, because that is to only non-standard thing we are doing.